pkg/controller/common.go

```diff
@@ -1259,3 +1320,64 @@ func containsString(key string, items []string) bool {
 	}
 	return false
 }
+
+type ReplicaRequest struct {
+	volumeID      string
+	volumeContext map[string]string //change this to include actual information
+	priority      int               //The number of replicas that have yet to be created
+	index         int
+}
+
+type VolumeReplicaRequestsPriorityQueue struct {
+	mut   *sync.Mutex
+	queue PriorityQueue
+}
+
+func (vq VolumeReplicaRequestsPriorityQueue) Push(replicaRequest ReplicaRequest) {
+	vq.mut.Lock()
+	defer vq.mut.Unlock()
+	vq.queue.Push(replicaRequest)
```

Comment:
This needs to be done through [`heap.Push()`](https://pkg.go.dev/container/heap#Push) in order to maintain the correct order, i.e.:
```go
heap.Push(&vq.queue, replicaRequest)
```